### PR TITLE
Retry status updates in case resource changes during reconciliation

### PR DESCRIPTION
**Pull request by ewrenn8**

### CODE REVIEW

File: pkg/packageinstall/packageinstall.go [hunk 1]
```diff
@@ -241,17 +241,28 @@ func (pi *PackageInstallCR) update(updateFunc func(*pkgingv1alpha1.PackageInstal
...
-	return nil
+		// if we errored, refresh the model we have
+		modelForUpdate, err = pi.kcclient.PackagingV1alpha1().PackageInstalls(modelForUpdate.Namespace).Get(
+			context.Background(), modelForUpdate.Name, metav1.GetOptions{})
+		if err != nil {
+			return fmt.Errorf("Refreshing package install model: %s", err)
+		}
+	}
+
+	return fmt.Errorf("Updating installed package: %s", lastErr)
```
Comment (danielhelfand) [inline pkg/packageinstall/packageinstall.go]:
```suggestion
	return fmt.Errorf("Updating PackageInstall: %s", lastErr)
```

File: pkg/packageinstall/packageinstall.go [hunk 2]
```diff
@@ -241,17 +241,28 @@ func (pi *PackageInstallCR) update(updateFunc func(*pkgingv1alpha1.PackageInstal
...
 
-	pi.unmodifiedModel = pi.model.DeepCopy()
+		lastErr = err
 
-	return nil
+		// if we errored, refresh the model we have
+		modelForUpdate, err = pi.kcclient.PackagingV1alpha1().PackageInstalls(modelForUpdate.Namespace).Get(
+			context.Background(), modelForUpdate.Name, metav1.GetOptions{})
+		if err != nil {
+			return fmt.Errorf("Refreshing package install model: %s", err)
```
Comment (danielhelfand) [inline pkg/packageinstall/packageinstall.go]: Refreshing is a bit confusing to me from the error standpoint.

```suggestion
			return fmt.Errorf("Getting PackageInstall model: %s", err)
```

File: pkg/packageinstall/packageinstall.go [hunk 3]
```diff
@@ -241,17 +241,28 @@ func (pi *PackageInstallCR) update(updateFunc func(*pkgingv1alpha1.PackageInstal
 	pi.log.Info("Updating installed package")
 
 	modelForUpdate := pi.model.DeepCopy()
-	updateFunc(modelForUpdate)
 
-	var err error
+	var lastErr error
+	for i := 0; i < 5; i++ {
```
Comment (danielhelfand) [inline pkg/packageinstall/packageinstall.go:246]: Any reason for 5 retries?

Comment (ewrenn8) [reply pkg/packageinstall/packageinstall.go:246]: No, it's just the number we went with last time we ran in to this

**Pull request merged**